core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -159,7 +166,7 @@ public TermsQueryBuilder(String fieldName, Iterable<?> values) {
             throw new IllegalArgumentException("No value specified for terms query");
         }
         this.fieldName = fieldName;
-        this.values = convertToBytesRefListIfStringList(values);
+        this.values = convert(values);
```

Comment:
can we add some dedicated tests to TermsQueryBuilderTest that stresses this entire convertion a bit? also with mixed value lists like floats / longs etc mixed